src/metrics/prometheus.go

```diff
@@ -158,7 +158,10 @@ func ExportPrometheusMetrics(ctx context.Context, gateways string) {
 	}
 	go func(ctx context.Context, server *http.Server) {
 		<-ctx.Done()
-		server.Shutdown(ctx)
+		err := server.Shutdown(ctx)
+		if err != nil {
+			log.Println("failure shutting down prometheus server:", err)
```

Comment:
The `http.ErrServerClosed`, that might be returned by the `server.Shutdown()` is not a failure.

Reply:
good point